provider/spark.go

```diff
@@ -716,7 +716,15 @@ func (store *SparkOfflineStore) GetBatchFeatures(ids []ResourceID) (BatchFeature
 	if err != nil {
 		return nil, fmt.Errorf("could not get output files: %v", err)
 	}
-	iterator, err := store.Store.Serve(outputFiles)
+	groups, err := filestore.NewFilePathGroup(outputFiles, filestore.DateTimeDirectoryGrouping)
```

Comment:
Do we need this?

Reply:
Since we only have one version

Reply:
We do need this given the UUID is deterministic given the name-variants and their order, so if we serve the same features in the same order, a new datetime directory will be written to `/featureform/BatchFeatures/<NAME>/<VARIANT>/`, so this grouping mechanism is key to avoid reading all parquet files from all materializations.